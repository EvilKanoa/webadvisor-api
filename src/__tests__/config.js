describe('Configuration controls', () => {
  let env;

  beforeAll(() => {
    env = process.env;
  });

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...env };
  });

  afterAll(() => {
    process.env = env;
  });

  it('uses the process.env values if set', () => {
    process.env = { NODE_ENV: 'test_env', PORT: 1337 };
    const config = require('../config');

    expect(JSON.stringify(config)).toMatchSnapshot();
  });

  it('uses fallbacks when process.env values are not set', () => {
    process.env = {};
    const config = require('../config');

    expect(JSON.stringify(config)).toMatchSnapshot();
  });
});
