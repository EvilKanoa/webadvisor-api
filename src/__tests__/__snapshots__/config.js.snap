// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration controls uses fallbacks when process.env values are not set 1`] = `"{\\"env\\":\\"production\\",\\"port\\":3001}"`;

exports[`Configuration controls uses the process.env values if set 1`] = `"{\\"env\\":\\"test_env\\",\\"port\\":1337}"`;
